### PR TITLE
[MRG+1] Remove copy from find_bads_exg.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3883

I couldn't find a reason we make a copy of the instance and drop a part of the channels as it seems to be handled elsewhere. This required the data to be preloaded. There's also check for this [here](https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/ica.py#L611).

Hopefully the tests will catch it if something broke.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Can you modify a test to ensure that the data is not modified with an `assert_array_equal` comparing the raw data before and after the processing? That would be safer. It's possible that there is some inplace filtering going on that existing tests wouldn't necessarily catch.

**Comment by jaeilepp** [comment]
I noticed that there was no tests for evoked instances so I added them too.

**Comment by Eric89GXL** [comment]
Did the tests pass locally?

Let me know when this is MRG and I'll have a look

**Comment by jaeilepp** [comment]
Test passed locally and looks like they're about to pass here. Ready for MRG.

**Comment by Eric89GXL** [comment]
LGTM +1 for merge

**Comment by agramfort** [comment]
@dengemann you wrote these initial copys maybe for a reason.

can you check this PR?

thx

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -1093,9 +1086,6 @@ def find_bads_eog(self, inst, ch_name=None, threshold=3.0,
         # get targets befor equalizing
         targets = [self._check_target(k, inst, start, stop) for k in eog_chs]
 
-        if inst.ch_names != self.ch_names:
-            inst = inst.copy().pick_channels(self.ch_names)
```
Comment (dengemann) [inline mne/preprocessing/ica.py]: I'm not sure out tests fully cover the consequences. The reason we need the copy here is that we need to pick in certain situations and then we don't want to modify the data that are passed here. If you want to change this you need to make sure you test well passing instances with different channels than ICA is fitted on and also deal with the ValueErrors we put here.

**Comment by dengemann** [comment]
Before merging let's add some more tests. The coverage of the tests here may not be excellent and the PR might change behavior in real world use cases. On should be able to run a script like this one  https://github.com/dengemann/meeg-preprocessing/blob/master/examples/plot_preprocess_filter_ica.py in which instances used to searching for artefacts can look different from the ones ICA was fitted on.

File: mne/preprocessing/ica.py (hunk 1 above)
Comment (jaeilepp) [inline mne/preprocessing/ica.py]: The tests already cover the case where the ica is fitted with picks. I also added tests where raw has channels dropped after fitting and raises an error. I can't imagine of any other use cases.

**Comment by dengemann** [comment]
Do we have a case where you have more channels on the inst than you have in
the fitted ICA? I think that's the relevant one. Let me run one of these
scripts later today to check for errors.

**Comment by jaeilepp** [comment]
> Do we have a case where you have more channels on the inst than you have in the fitted ICA?

Yes. See the latest commit. I meant to add it earlier, but I forgot the actual call to the function.

**Comment by dengemann** [comment]
Ok, I'll merge after running some tests in a free moment today.

**Comment by dengemann** [comment]
looks good on my end, merging. Thank you Jaakko!

**Pull request merged**